streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -327,15 +331,15 @@ private void prepareStreamThread(final StreamThread thread, final boolean termin
 
     @Test
     public void testShouldTransitToNotRunningIfCloseRightAfterCreated() {
-        final KafkaStreams streams = new KafkaStreams(new StreamsBuilder().build(), props, supplier, time);
+        final KafkaStreams streams = new KafkaStreams(getBuilderWithSource().build(), props, supplier, time);
```

Comment:
The change to reject an empty topology make our testing rather "ugly"... Do we really _need_ to reject it?

Reply:
Is `getBuilderWithSource` really that much uglier than `new StreamsBuilder`? :P

Reply:
Would a class-level `final StreamsBuilder builder = new StreamsBuilder()` that we add a source to in the setUp be any. better iyo?

Reply:
I think the suggestion was more along the lines of _not_ throwing an exception while building an empty topology.

I'm not sure. It seems kind of nice to find out right away that your program will do absolutely nothing.

Reply:
I realize he wasn't just complaining about the name, but I was trying to keep that discussion in one thread. But I guess you can only do so much to keep PR chatter oriented in one place

Reply:
But @vvcephei 's last question gets right to the heart of the matter. The answer being " technically yes, but it will crash if you try to poll for said nothing, so really no"
That's why the test was flaky, and the reason for this PR in the first place (Avoiding group overhead is the name of the ticket, but the reality is it will only happen once before all the StreamThreads die due to polling no topics)

Reply:
Thanks for clarification.